# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.cpp
# Commit: 77b0f5226ed8
# Full Hash: 77b0f5226ed854c6b8d1b8d13f3106c12ad0701f
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 2
# Description:
#   Bug 1831451 - Add a blank VideoDecoder interface r=webidl,padenot,edgar,smaug,saschanaz
#   
#   This patch introduces a VideoDecoder interface generated by running
#   `./mach webidl-example VideoDecoder` with necessary changes to make it
#   buildable.
# ==============================================================================

diff -r fb3b06cc9188 -r 77b0f5226ed8 dom/media/webcodecs/VideoDecoder.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/media/webcodecs/VideoDecoder.cpp	Thu Jun 29 22:33:11 2023 +0000
@@ -0,0 +1,71 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim:set ts=2 sw=2 sts=2 et cindent: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "mozilla/dom/VideoDecoder.h"
+#include "mozilla/dom/VideoDecoderBinding.h"
+
+namespace mozilla::dom {
+
+NS_IMPL_CYCLE_COLLECTION_INHERITED(VideoDecoder, DOMEventTargetHelper)
+NS_IMPL_ADDREF_INHERITED(VideoDecoder, DOMEventTargetHelper)
+NS_IMPL_RELEASE_INHERITED(VideoDecoder, DOMEventTargetHelper)
+NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(VideoDecoder)
+NS_INTERFACE_MAP_END_INHERITING(DOMEventTargetHelper)
+
+JSObject* VideoDecoder::WrapObject(JSContext* aCx,
+                                   JS::Handle<JSObject*> aGivenProto) {
+  return VideoDecoder_Binding::Wrap(aCx, this, aGivenProto);
+}
+
+/* static */
+already_AddRefed<VideoDecoder> VideoDecoder::Constructor(
+    const GlobalObject& global, const VideoDecoderInit& init,
+    ErrorResult& aRv) {
+  aRv.Throw(NS_ERROR_DOM_NOT_SUPPORTED_ERR);
+  return nullptr;
+}
+
+CodecState VideoDecoder::State() const { return CodecState::EndGuard_; }
+
+uint32_t VideoDecoder::DecodeQueueSize() const { return 0; }
+
+already_AddRefed<EventHandlerNonNull> VideoDecoder::GetOndequeue() const {
+  return nullptr;
+}
+
+void VideoDecoder::SetOndequeue(EventHandlerNonNull* arg) {}
+
+void VideoDecoder::Configure(const VideoDecoderConfig& config,
+                             ErrorResult& aRv) {
+  aRv.Throw(NS_ERROR_DOM_NOT_SUPPORTED_ERR);
+}
+
+void VideoDecoder::Decode(EncodedVideoChunk& chunk, ErrorResult& aRv) {
+  aRv.Throw(NS_ERROR_DOM_NOT_SUPPORTED_ERR);
+}
+
+already_AddRefed<Promise> VideoDecoder::Flush(ErrorResult& aRv) {
+  aRv.Throw(NS_ERROR_DOM_NOT_SUPPORTED_ERR);
+  return nullptr;
+}
+
+void VideoDecoder::Reset(ErrorResult& aRv) {
+  aRv.Throw(NS_ERROR_DOM_NOT_SUPPORTED_ERR);
+}
+
+void VideoDecoder::Close(ErrorResult& aRv) {
+  aRv.Throw(NS_ERROR_DOM_NOT_SUPPORTED_ERR);
+}
+
+/* static */
+already_AddRefed<Promise> VideoDecoder::IsConfigSupported(
+    const GlobalObject& global, const VideoDecoderConfig& config,
+    ErrorResult& aRv) {
+  aRv.Throw(NS_ERROR_DOM_NOT_SUPPORTED_ERR);
+  return nullptr;
+}
+
+}  // namespace mozilla::dom